src/main/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJob.java

```diff
@@ -33,23 +37,50 @@ public HttpCallJob(
     }
 
     @Override
-    public void execute(JobExecutionContext context) {
+    public void execute(JobExecutionContext context) throws JobExecutionException {
         String jobId = context.getJobDetail().getKey().getName();
-        logger.info("Executing job " + jobId);
 
-        HttpAction action = actionExtractor.extract(context)
-            .withHeader("ServiceAuthorization", tokenGenerator.generate());
+        try {
+            logger.info("Executing job " + jobId);
 
-        ResponseEntity<String> response =
-            restTemplate
-                .exchange(
-                    action.url,
-                    action.method,
-                    toHttpEntity(action),
-                    String.class
-                );
+            HttpAction action = actionExtractor.extract(context)
+                .withHeader("ServiceAuthorization", tokenGenerator.generate());
 
-        logger.info("Job {} executed. Response code: {}", jobId, response.getStatusCodeValue());
+            ResponseEntity<String> response =
+                restTemplate
+                    .exchange(
+                        action.url,
+                        action.method,
+                        toHttpEntity(action),
+                        String.class
+                    );
+
+            logger.info("Job {} executed. Response code: {}", jobId, response.getStatusCodeValue());
+
+            failIfErrorResponse(response, jobId);
```

Comment:
we could move this statement to the end of function block and get rid of extra catch statement